### DISCUSSION TITLE
Query param alias generator

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#4070] [Questions]
Is there a workaround to get query_param name and camel_case it and add it as an alias?

**Comment by ghandic** [comment]
Unsure what youre asking here, could you add an example?

**Comment by ghost** [comment]
@ghandic 
So I found out Camelcasing the Pydantic keys dynamically
Is there a way to do it with params that we do in FastApi?
E.g.
```
@app.get("/fields")
async def fields(variable_one: dict = None,
                 variable_two: int = Query(None, gt=0),
                 variable_three: int = Query(None, gt=0)):
    pass
```
How do I dynamically add camelcase alias to my params, is the Question?

**Reply by ghandic** [reply]
You can use alias and create a decorator to modify a function to modify the signature to add camel case alias' to all query params.

If you just want it as a one off I'd just use alias directly https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#alias-parameters

**Reply by ghandic** [reply]
Also see some good utils to do the camel casing from @dmontagu 's fastapi utils package https://fastapi-utils.davidmontague.xyz/user-guide/basics/camelcase/

**Reply by ghost** [reply]
@ghandic Thanks for the quick response

1. It's not a one-off so I think I'll skip adding it manually for all fields.
2. The Second method using the utils works on pydantic model definition which are not used on params, it can be used for request body.
3. It would be great if you can give an example of a decorator definition that can modify the signature.

Thanks again for the help 😄

**Comment by HansBrende** [comment]
Using a decorator would be too much boilerplate IMO... here's how I solved this problem for my api:

```python
app: FastAPI = ...

# add routes and/or routers first...
app.include_router(...)

for route in app.routes:
    if isinstance(route, APIRoute):
        for param in route.dependant.query_params:
            if not param.field_info.alias:
                param.alias = camelcase(param.name)
```

**Reply by nameer** [reply]
Nice. However, `param.field_info.alias = camelcase(param.name)` is what worked for me in FastAPI version `0.104.1`.

**Comment by glebignatieff** [comment]
I'll add my two cents here. I'm trying to use camel case for the external API while using snake case internally. For this I have this model, that I inherit from:

```python
class BaseModel(pydantic.BaseModel):
    model_config = pydantic.ConfigDict(
        alias_generator=alias_generators.to_camel,
    )
```

The problem is that for query param models the generator doesn't work as expected. Say I have the following model:

```python
class Filters(BaseModel):
    file_type: str
```

I add it as instructed in the [docs](https://fastapi.tiangolo.com/tutorial/query-param-models/):

```python
filters: t.Annotated[models.Filters, fastapi.Query()],
```

It's displayed correctly in the docs:

<img width="133" alt="image" src="https://github.com/user-attachments/assets/f52bfca4-0ce2-472b-b5f5-3cda48554e1d" />

However, when I make a request with `fileType=something`, I get the following error:

```shell
Validation error: [{'type': 'missing', 'loc': ('query', 'fileType'), 'msg': 'Field required', 'input': {'file_type': 'something'}}]
```
